lib/cambiatus_web/schema/shop_types.ex

```diff
@@ -29,6 +29,13 @@ defmodule CambiatusWeb.Schema.ShopTypes do
       middleware(Middleware.Authenticate)
       resolve(&Shop.get_product/3)
     end
+
+    @desc "Public product query, mainly used to preview a sale"
+    field(:product_preview, :product_preview) do
+      arg(:id, non_null(:integer))
+
+      resolve(&Shop.get_product/3)
+    end
```

Comment:
Can we always return a `:product_preview` here? If the product doesn't exist, the frontend receives a GraphQL error anyway, so I guess it doesn't make sense to have a `Maybe ProductPreview` on the frontend. (The same applies to the `product` query)

```suggestion
    @desc "Public product query, mainly used to preview a sale"
    field(:product_preview, non_null(:product_preview)) do
      arg(:id, non_null(:integer))

      resolve(&Shop.get_product/3)
    end
```

Reply:
opa! my bad bro

this default to null is extra clumsy intuitive to me